### PR TITLE
Ruby: Add some method calls as a Source

### PR DESCRIPTION
**Pull request by Sim4n6**

### CODE REVIEW

File: ruby/ql/lib/codeql/ruby/experimental/UnicodeBypassValidationQuery.qll [hunk 1]
```diff
@@ -79,6 +79,47 @@ private module UnicodeBypassValidationConfig implements DataFlow::StateConfigSig
...
+            [
+                "ljust", "lstrip", "succ", "next", "rjust", "capitalize", "chomp", "gsub", "chop",
+                "downcase", "swapcase", "uprcase", "scrub", "slice", "squeeze", "strip", "sub",
+                "tr", "tr_s", "reverse"
+              ] + ["", "!"], "concat", "dump", "each_line", "replace", "insert", "inspect", "lines",
+            "partition", "prepend", "replace", "rpartition", "scan", "split", "undump",
+            "unpack" + ["", "1"]
+          ] and
+        source = cn and
+        source.getLocation().getFile().getBaseName().matches("object.rb")
```
Comment (aibaars) [inline ruby/ql/lib/codeql/ruby/experimental/UnicodeBypassValidationQuery.qll:104]: Why match the file name against `object.rb` ?

**Review by aibaars — COMMENTED** [review]
This looks good to me, just wondering about the match against `objects.rb`.

File: ruby/ql/lib/codeql/ruby/experimental/UnicodeBypassValidationQuery.qll (hunk 1 above)
Comment (Sim4n6) [reply ruby/ql/lib/codeql/ruby/experimental/UnicodeBypassValidationQuery.qll:104]: À mistake for sure, no need to limit against object.rb file.

File: ruby/ql/lib/codeql/ruby/experimental/UnicodeBypassValidationQuery.qll [hunk 2]
```diff
@@ -79,6 +79,46 @@
 
   predicate isSource(DataFlow::Node source, FlowState state) {
     source instanceof RemoteFlowSource and state = PreValidationState()
+    or
+    (
+      exists(Escaping escaping | source = escaping.getOutput())
+      or
+      exists(RegexExecution re | source = re)
```
[the remark this answers is not in the document]
Comment (Sim4n6) [reply ruby/ql/lib/codeql/ruby/experimental/UnicodeBypassValidationQuery.qll]: fixed in commit https://github.com/github/codeql/pull/16471/commits/dabc33bf6675bd26e157cb803c2653abc9ba67ce

**Comment by aibaars** [comment]
@Sim4n6 There seems to be a change in the expected output from the tests. Could you verify and update the test data if needed?

**Comment by Sim4n6** [comment]
Hey, 

Well, I don't know what is the issue right away. So I updated the Codeql cli to 2.17.4 and I rerun the codeql test learn to regenerate the expected file. 

I am available right away to fix this issue, please explain more thank you

**Comment by Sim4n6** [comment]
hey @aibaars , any further improvements that I should do related to this one  ?

**Review by aibaars — APPROVED** [review]
Looks good to me, thanks!

**Pull request merged**